app/src/main/java/com/example/instagramclone/fragments/main_screen/compose/PostItem.kt

```diff
@@ -0,0 +1,215 @@
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
+                text = "2,426" + " likes",
+                fontWeight = FontWeight.Bold
+            )
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
+                text = buildAnnotatedString {
+                    withStyle(style = SpanStyle(fontWeight = FontWeight.Bold)) {
+                        append(postItem.userName)
+                        append(" ")
+                    }
+                    append(postItem.contentDesc)
+                }
+            )
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
+                text = "View all 83 comments",
+                style = TextStyle(color = Color.Gray)
+            )
+            Text(
+                modifier = modifier
+                    .padding(horizontal = 15.dp),
+                text = "4 Hours ago",
+                style = TextStyle(color = Color.Gray)
+            )
+        }
+    }
+}
+
+@Composable
+fun PostBottomBar(savePost: () -> Unit) {
+    var isPostLiked by remember {
+        mutableStateOf(false)
+    }
+    var isPostSaved by remember {
+        mutableStateOf(false)
+    }
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(10.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        Row() {
+            BottomBarIcon(
+                onClick = {
+                    isPostLiked = !isPostLiked
+                },
+                painter = painterResource(
+                    id = if (isPostLiked) {
+                        drawable.ic_heart_64_true
+                    } else {
+                        drawable.ic_heart_64_false
+                    }
+                )
+            )
+            BottomBarIcon(
+                onClick = { },
+                painter = painterResource(id = drawable.ic_comment)
+            )
+
+            BottomBarIcon(
+                onClick = { },
+                painter = painterResource(id = drawable.ic_share)
+            )
+        }
+        BottomBarIcon(
+            onClick = {
+                isPostSaved = !isPostSaved
+                savePost()
+            },
+            painter = painterResource(
+                id = if (isPostSaved) {
+                    drawable.ic_save_true
+                } else {
+                    drawable.ic_save_false
+                }
+            )
+        )
+    }
+}
+
+@Composable
+fun BottomBarIcon(
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+    painter: Painter
+) {
+    Icon(
+        modifier = modifier
+            .padding(horizontal = 8.dp)
+            .clickable(interactionSource = MutableInteractionSource(), indication = null) {
+                onClick()
+            },
```

Comment:
Optional: This could also be written as `.clickable(interactionSource = MutableInteractionSource(), indication = null, onClick=onClick)`